aspnetcore/host-and-deploy/aspnet-core-module.md

```diff
@@ -13,12 +13,97 @@ By [Luke Latham](https://github.com/guardrex), [Rick Anderson](https://twitter.c
+
+To configure an app for in-process hosting, add the `<AspNetCoreModuleHostingModel>` property to the app's project file with a value of `inprocess` (out-of-process hosting is set with `outofprocess`):
+
+```xml
+<PropertyGroup>
+  <AspNetCoreModuleHostingModel>inprocess</AspNetCoreModuleHostingModel>
+</PropertyGroup>
+```
+
+The following characteristics apply when hosting in-process:
+
+* The [Kestrel server](xref:fundamentals/servers/kestrel) isn't used. A custom <xref:Microsoft.AspNetCore.Hosting.Server.IServer> implementation, `IISHttpServer` acts as the app's server.
+
+* The [requestTimeout attribute](#attributes-of-the-aspnetcore-element) doesn't apply to in-process hosting.
+
+* Sharing an app pool among apps isn't supported. Use one app pool per app.
+
+* When using [Web Deploy](/iis/publish/using-web-deploy/introduction-to-web-deploy) or manually placing an [app_offline.htm file in the deployment](xref:host-and-deploy/iis/index#locked-deployment-files), the app might not shut down immediately if there's an open connection. For example, a websocket connection may delay app shut down.
+
+* The architecture (bitness) of the app and installed runtime (x64 or x86) must match the architecture of the app pool.
+
+* If setting up the app's host manually with `WebHostBuilder` (not using [CreateDefaultBuilder](xref:fundamentals/host/web-host#set-up-a-host)) and the app is ever run directly on the Kestrel server (self-hosted), call `UseKestrel` before calling `UseIISIntegration`. If the order is reversed, the host fails to start.
+
+* Don't use <xref:System.IO.Directory.GetCurrentDirectory*> to set the content root with <xref:Microsoft.Extensions.Hosting.HostingHostBuilderExtensions.UseContentRoot*>. See the [Content Root](#content-root) section for details.
+
+### Hosting model changes
+
+If the `hostingModel` setting is changed in the *web.config* file (explained in the [Configuration with web.config](#configuration-with-webconfig) section), the module recycles the worker process for IIS.
+
+For IIS Express, the module doesn't recycle the worker process but instead triggers a graceful shutdown of the current IIS Express process. The next request to the app spawns a new IIS Express process.
+
+### Content root
+
+When running an app in-process, the current directory reported by <xref:System.IO.Directory.GetCurrentDirectory*> is *C:\\Windows\\System32\\inetsrv\\*. To set the content root, use [AppDomain.CurrentDomain.BaseDirectory](xref:System.AppDomain.BaseDirectory*) with <xref:Microsoft.Extensions.Hosting.HostingHostBuilderExtensions.UseContentRoot*>:
+
+```csharp
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        CreateWebHostBuilder(args).Build().Run();
+    }
+
+    public static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
+        WebHost.CreateDefaultBuilder(args)
+            .UseContentRoot(AppDomain.CurrentDomain.BaseDirectory)
```

Comment:
@pakrym can UseIIS do this for you?

Reply:
It already does: https://github.com/aspnet/IISIntegration/blob/afa5d60821eef1731482bafd1134a4cfbc05eb48/src/Microsoft.AspNetCore.Server.IIS/WebHostBuilderIISExtensions.cs#L38

Reply:
Great, this section can go.